pkg/helpers/helpers.go

```diff
@@ -528,3 +536,37 @@ func GetNodeSelector(cluster *clusterv1.ManagedCluster) (map[string]string, erro
 
 	return nodeSelector, nil
 }
+
+// DetermineKlusterletMode gets the klusterlet deploy mode for the managed cluster.
+func DetermineKlusterletMode(cluster *clusterv1.ManagedCluster) (string, error) {
```

Comment:
nit: how about only return the mode, use another method to handle `GetManagementCluster`

Reply:
for other cases, return `unknown`